modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageManagerImpl.java

```diff
@@ -121,12 +105,17 @@ public class MetaStorageManagerImpl implements MetaStorageManager {
     /** Prevents double stopping of the component. */
     private final AtomicBoolean isStopped = new AtomicBoolean();
 
+    private final AtomicLong appliedRevision = new AtomicLong(0L);
+
     /**
      * Future which completes when MetaStorage manager finished local recovery. The value of the future is the revision which must be used
      * for state recovery by other components.
      */
     private final CompletableFuture<Long> recoveryFinishedFuture = new CompletableFuture<>();
 
+    private final CompletableFuture<Long> recoveryFinishedPublicFuture
+            = recoveryFinishedFuture.whenComplete((revision, e) -> appliedRevision.set(revision));
```

Comment:
maybe you intended to use `thenAccept` here?

Reply:
Maybe

Reply:
Can `recoveryFinishedFuture` be completed with a exception? If yes, then the current approach is incorrect (you should verify that `e != null`)